crates/bevy_ecs/src/schedule/executor/simple.rs

```diff
@@ -78,9 +79,15 @@ impl SystemExecutor for SimpleExecutor {
             let system = &mut schedule.systems[system_index];
             #[cfg(feature = "trace")]
             let system_span = info_span!("system", name = &*name).entered();
-            system.run((), world);
+            let res = std::panic::catch_unwind(AssertUnwindSafe(|| {
+                system.run((), world);
+            }));
             #[cfg(feature = "trace")]
             system_span.exit();
+            if let Err(payload) = res {
+                eprintln!("Encountered a panic in system `{}`!", &*system.name());
```

Comment:
It's a bit too late now, but in a future PR something like this could make the error stand out a bit more

```suggestion
                eprintln!("\x1b[91mERROR\x1b[0m: Encountered a panic in system `{}`!", &*system.name());
```

This will make the ERROR red and look like this:
![image](https://user-images.githubusercontent.com/8348954/231554647-72308de0-dada-4f23-a88c-3d89f4b33130.png)

Reply:
is this supported in all terminals on all platforms?

Reply:
I know it works on windows at least and I would be surprised if it didn't work on linux. but I don't have a way to confirm it right now. There are crates that can handle that more gracefully though, just not sure if adding a crate just for that is required, although, tracing probably already uses one of them so we could just reuse that one since it's already in the dep tree.

Reply:
And when logs are redirected to a file? I don't know of a single way to have colors...

Reply:
We could use `error!` macro to have colors in terminal and normal output if colors aren't supported.

Reply:
> We could use error! 

`error!` wasn't used because of verbosity concerns. I don't personally agree, but I suggested that approach as an alternative. I think we should instead change the default format of logs if the verbosity is a concern instead of avoiding it, but that's out of scope for here.

> And when logs are redirected to a file

True, I didn't really consider that. At least it would just be a couple of weird characters at the end of the file, but it's indeed not ideal.

Reply:
> I don't know of a single way to have colors...

Well, there's `error!()` that can deal with it correctly 😉. So it's _a_ way, but like mentioned it was purposefully not used.

Reply:
> Well, there's error!() that can deal with it correctly

I meant by adding control characters. As far as I know the tracing library is using environment detection.

> `error!` wasn't used because of verbosity concerns

Oh I thought it was because `bevy_eccs` can be used without logs and tracing enabled